java/src/org/openqa/selenium/docker/v1_41/CreateContainer.java

```diff
@@ -52,6 +53,24 @@ public CreateContainer(DockerProtocol protocol, HttpHandler client) {
   }
 
   public Container apply(ContainerConfig info) {
+    try {
+      return createContainer(info);
+    } catch (DockerException e) {
+      if (e.getMessage().contains("No such image")) {
+        return pullImageAndCreateContainer(info);
+      } else {
+        throw e;
+      }
```

Comment:
Could we take an approach where we check if the image exists locally first? If not, we pull it. 

I prefer to avoid checking exception strings.

Reply:
@diemol, after exploring methods, I see that need to call a single method `getImage()` from `DockerProtocol`, check and pull image if missing is taking care in there.